sentry-android-timber/src/main/java/io/sentry/android/timber/SentryTimberTree.kt

```diff
@@ -0,0 +1,102 @@
+package io.sentry.android.timber
+
+import android.util.Log
+import io.sentry.core.Breadcrumb
+import io.sentry.core.IHub
+import io.sentry.core.SentryEvent
+import io.sentry.core.SentryLevel
+import io.sentry.core.protocol.Message
+import timber.log.Timber
+
+/**
+ * Sentry Timber tree which is responsible to capture events via Timber
+ */
+class SentryTimberTree(
+    private val hub: IHub,
+    private val minEventLevel: SentryLevel,
+    private val minBreadcrumbLevel: SentryLevel
+) : Timber.Tree() {
+
+    /**
+     * do not log if it's lower than min. required level.
+     */
+    private fun isLoggable(level: SentryLevel, minLevel: SentryLevel): Boolean = level.ordinal >= minLevel.ordinal
+
+    /**
+     * Compares the minLevel given by the user and the level given by the event
+     * to capture an event or not.
+     */
+    override fun isLoggable(tag: String?, priority: Int): Boolean {
+        val level = getSentryLevel(priority)
+
+        // checks only the event level
+        return isLoggable(level, minEventLevel)
+    }
+
+    /**
+     * Captures a Sentry Event if the min. level is equal or higher than the min. required level.
+     */
+    override fun log(priority: Int, tag: String?, message: String, throwable: Throwable?) {
+        val level = getSentryLevel(priority)
+
+        captureEvent(level, tag, message, throwable)
+        addBreadcrumbIfEventHasThrowable(level, throwable, message)
+    }
+
+    /**
+     * Captures an event with the given attributes
+     */
+    private fun captureEvent(level: SentryLevel, tag: String?, message: String, throwable: Throwable?) {
+        val sentryEvent = SentryEvent()
+        sentryEvent.level = level
+
+        // if there's no throwable, should it be a breadcrumb then?
+        throwable?.let {
+            sentryEvent.setThrowable(it)
+        }
+        sentryEvent.message = Message().apply {
+            formatted = message
+        }
+
+        tag?.let {
```

Comment:
L: just realize we could use the event.logger field here instead of a custom tag

Reply:
the way how it works on Timber is different. see:

> Set a one-time tag for use on the next logging call

So I'd rather keep this tag which is a custom tag set by the client, but also setting event.logger always to Timber.